packages/@glimmer/manager/lib/internal/index.ts

```diff
@@ -89,12 +97,16 @@ export function getInternalModifierManager(definition: object): InternalModifier
 
   const manager = getManager(MODIFIER_MANAGERS, definition)!;
 
-  if (DEBUG && manager === undefined) {
-    throw new Error(
-      `Attempted to load a modifier, but there wasn't a modifier manager associated with the definition. The definition was: ${debugToString!(
-        definition
-      )}`
-    );
+  if (manager === undefined) {
+    if (isOptional === true) {
+      return null;
```

Comment:
What is `isOptional` for conceptually?

Reply:
For components and helpers, it's actually valid for a value to not be a component or a helper, but we still have to check if it is one (e.g. in append position, `{{maybeComponentOrHelper}}`), so that's why they have this capability. For modifiers, I added it so we can throw a better error message for dynamic modifiers specifically, because we have more context (what path the modifier was invoking from).